pkg/storage/kubernetes.go

```diff
@@ -86,7 +86,9 @@ func (k kubernetesClient) Delete(name string) error {
 func (k kubernetesClient) List() (StorageList, error) {
 	if k.deployment == nil {
 		var err error
-		k.deployment, err = k.client.GetOneDeployment(k.componentName, k.appName)
+		// FIXME: Find a better way to add the component label to selector
+		selector := fmt.Sprintf("%s,component=%s", odolabels.GetSelector(k.componentName, k.appName, odolabels.ComponentAnyMode), k.componentName)
```

Comment:
Same comment as before: Would a better way be to add a new boolean to the labels.GetSelector function to indicate whether it should include the component label as well?